learning/tour-of-beam/frontend/lib/pages/tour/widgets/group.dart

```diff
@@ -41,37 +41,56 @@ class GroupWidget extends StatelessWidget {
       builder: (context, child) {
         final isExpanded = contentTreeController.expandedIds.contains(group.id);
 
-        return ExpansionTileWrapper(
-          ExpansionTile(
-            key: Key('${group.id}$isExpanded'),
-            initiallyExpanded: isExpanded,
-            tilePadding: EdgeInsets.zero,
-            onExpansionChanged: (isExpanding) {
-              /// Since expandedIds is also used for expansion,
-              /// it needs to be updated to prevent the tile from
-              /// remaining collapsed after taps on title.
-              if (!isExpanding) {
-                contentTreeController.markGroupAsCollapsed(group);
-              }
-            },
-            title: GroupTitleWidget(
-              group: group,
-              onTap: () {
-                contentTreeController.openNode(group);
-              },
-            ),
-            childrenPadding: const EdgeInsets.only(
-              left: BeamSizes.size24,
-            ),
-            children: [
-              GroupNodesWidget(
-                nodes: group.nodes,
-                contentTreeController: contentTreeController,
-              ),
-            ],
-          ),
+        return _StatelessExpansionTile(
+          contentTreeController: contentTreeController,
+          group: group,
+          isExpanded: isExpanded,
         );
       },
     );
   }
 }
+
+class _StatelessExpansionTile extends StatelessWidget {
```

Comment:
I had an idea of a generic `StatelessExpensionTile` that has no idea of group and controller, and would use most of `ExpansionTile` parameters except that it is guided by an external boolean state. It could then be reusable. Not that we plan to reuse it soon, but its use is just cleaner because then we would not have `key`.

This change adds an ad-hoc widget to the extent that it is private. In this context it feels more of a cluttering than a benefit. For instance, it has `isExpanded` which is a function of its other two parameters.

I suggest either reverting this change, or going the full way.